python/dgl/graphbolt/impl/csc_sampling_graph.py

```diff
@@ -362,7 +362,7 @@ def _convert_to_homogeneous_nodes(self, nodes):
         for ntype, ids in nodes.items():
             ntype_id = self.metadata.node_type_to_id[ntype]
             homogeneous_nodes.append(ids + self.node_type_offset[ntype_id])
-        return torch.cat(homogeneous_nodes)
+        return torch.cat(homogeneous_nodes).to(self.csc_indptr.dtype)
```

Comment:
Ideally, we don't need to convert the data type during sampling, the dataset should use the correct dtype already.

Reply:
Usually users are not aware of(even not care about it at all) the dtype requirement when working on self-curated dataset. Adding such dtype formatting is just an additional guard.

Reply:
such format is removed as we don't need it any more in the latest implementation. please let me know if you have any more concerns.